modules/roles/manifests/joechem.pp

```diff
@@ -0,0 +1,63 @@
+class roles::joechem {
+    class { '::php':
+      ensure       => latest,
+      manage_repos => true,
+      fpm          => true,
+      dev          => true,
+      composer     => true,
+      pear         => true,
+      phpunit      => false,
+      extensions   => {
+                curl => {},
+                pdo => {},
+              mysql => {}
+      }
+    }
+    ~>
+    class { 'apache':
+      default_mods  => true,
+      default_vhost => false,
+      # This defaults to 'worker' on debian, setting to false 
+      # allows specifying our own.
+      mpm_module    =>  false,
+    }
+
+    include apache::mod::prefork
+    include apache::mod::php
+    include apache::mod::rewrite
+
+    unless $facts['ec2userdata'] 
```

Comment:
just fyi you could use `$::ec2userdata`.

Reply:
I only did that cause I saw that varied between some versions of Puppet (I think), so I just wanted to be safe.